modules/ui/sections/background_list.js

```diff
@@ -61,6 +61,77 @@ export function uiSectionBackgroundList(context) {
       .append('ul')
       .attr('class', 'layer-list bg-extras-list');
 
+    const wayBackImageryEnter = bgExtrasListEnter
+      .append('li')
+      .attr('class', 'background.wayback_imagery.tooltip')
+      .append('div')
+      .style('display', 'flex');
+
+    const label = wayBackImageryEnter
+    .append('label')
+    .call(uiTooltip(context)
+      .title(l10n.t('background.wayback_imagery.tooltip'))
+      .placement('top')
+    );
+
+    label
+      .append('input')
+      .attr('type', 'checkbox')
+      .on('change', function(d3_event) {
+        d3_event.preventDefault();
+        dropdown.style('display', this.checked ? 'block' : 'none');
+
+        if (this.checked) {
+          imagery.getWaybackSource().then(sourceMap => {
+            const sources = Array.from(sourceMap.sources.values());
+            const dates = sources.map(source => {
+              const date = source.id.split('_')[1];
+              return { value: date };
+            });
+            dates.sort((a, b) => b.value.localeCompare(a.value));
+            dates.forEach(date => {
+              dropdown.append('option').text(date.value);
+            });
+            dropdown.node().dispatchEvent(new Event('change'));
+          });
+        } else {
+          dropdown.selectAll('option').remove();
+          const defaultSource = imagery.chooseDefaultSource();
+          imagery.baseLayerSource(defaultSource);
+        }
+      });
+
+    const dropdown = wayBackImageryEnter
+      .append('select')
+      .style('display', 'none')
+      .style('vertical-align', 'middle')
+      .style('height', '20px')
+      .style('width', '120px')
+      .style('margin', 'auto')
+      .style('cursor', 'pointer')
+      .on('change', function() {
```

Comment:
There's two `.on('change',` event handlers declared here and on line 120- they both seem to be doing the same thing. Is this intended?

Reply:
There were two main tasks: populating the dropdown with dates, and changing the imagery when a new date was selected. I've cleaned it up a bit for less repeated code